flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/AsyncCheckpointRunnable.java

```diff
@@ -110,8 +109,6 @@ public void run() {
         FileSystemSafetyNet.initializeSafetyNetForThread();
         try {
 
-            registerConsumer.accept(this);
```

Comment:
Is this change necessary? We are changing the time when the runnable is registered slightly. Before the change it was registered when run, now when the runnable is created.

Reply:
I think it might be necessary since otherwise there might be the case

1. We trigger a checkpoint and schedule its asynchronous part.
2. We call `waitForPendingCheckpoints()`. However, the asynchronous part is not executed yet.
3. After the call is done, the asynchronous part is executed and register the runnable. 

Then this runnable would be missed~

Reply:
BTW, sorry I missed one comment in the previous review (not sure how). Let's inline the regsiterConsumer() and remove the method. It is unnecessary to go through the `Consumer` here.